src/main/java/com/algorand/algosdk/transaction/AtomicTransactionComposer.java

```diff
@@ -417,36 +417,37 @@ public ExecuteResult execute(AlgodClient client, int waitRounds) throws Exceptio
             if (!this.methodMap.containsKey(i))
                 continue;
 
-            if (this.methodMap.get(i).returns.type.equals(Method.Returns.VoidRetType)) {
-                retList.add(new ReturnValue(
-                        this.transactionList.get(i).txn.txID(),
-                        new byte[]{},
-                        null,
-                        null
-                ));
-                continue;
-            }
-
-            PendingTransactionResponse respBody = txInfo;
-
+            PendingTransactionResponse currentTxInfo = txInfo;
             if (i != indexToWait) {
                 Response<PendingTransactionResponse> resp =
-                        client.PendingTransactionInformation(this.transactionList.get(i).txn.txID()).execute();
+                        client.PendingTransactionInformation(signedTxns.get(i).transactionID).execute();
                 if (!resp.isSuccessful()) {
                     retList.add(new ReturnValue(
+                            signedTxns.get(i).transactionID,
                             null,
                             null,
-                            null,
-                            new Exception(resp.message())
+                            new Exception(resp.message()),
+                            resp.body()
```

Comment:
Won't the call to `resp.body()` fail if the response was not successful?

Reply:
No, it would return null. Shall I just say null here?

Reply:
Ah ok. I think putting null would be clearer